operator/exporter.py

```diff
@@ -0,0 +1,371 @@
+        storage_metrics.cpu_usage.labels(
+            nodeplugin["pod_name"]).set(nodeplugin["cpu_usage_in_nanoseconds"])
+
+        storage_metrics.total_number_of_containers.labels(
+            nodeplugin["pod_name"]).set(nodeplugin["total_number_of_containers"])
+        storage_metrics.number_of_ready_containers.labels(
+            nodeplugin["pod_name"]).set(nodeplugin["number_of_ready_containers"])
+
+    # Storage(s) Metrics
+    for storage in metrics.storages:
+        storage_metrics.total_capacity_bytes.labels(storage["name"]).set(storage["total_capacity_bytes"])
+        storage_metrics.used_capacity_bytes.labels(storage["name"]).set(storage["used_capacity_bytes"])
+        storage_metrics.free_capacity_bytes.labels(storage["name"]).set(storage["free_capacity_bytes"])
+
+        storage_metrics.total_inodes.labels(storage["name"]).set(storage["total_inodes"])
+        storage_metrics.used_inodes.labels(storage["name"]).set(storage["used_inodes"])
+        storage_metrics.free_inodes.labels(storage["name"]).set(storage["free_inodes"])
+
+        for pvc in storage["pvc"]:
+            storage_metrics.total_pvc_capacity_bytes.labels(pvc["pvc_name"]).set(pvc["total_pvc_capacity_bytes"])
+            storage_metrics.used_pvc_capacity_bytes.labels(pvc["pvc_name"]).set(pvc["used_pvc_capacity_bytes"])
+            storage_metrics.free_pvc_capacity_bytes.labels(pvc["pvc_name"]).set(pvc["free_pvc_capacity_bytes"])
+
+            storage_metrics.total_pvc_inodes.labels(pvc["pvc_name"]).set(pvc["total_pvc_inodes"])
+            storage_metrics.used_pvc_inodes.labels(pvc["pvc_name"]).set(pvc["used_pvc_inodes"])
+            storage_metrics.free_pvc_inodes.labels(pvc["pvc_name"]).set(pvc["free_pvc_inodes"])
+
+        for brick in storage["bricks"]:
+            storage_metrics.memory_usage.labels(
+                brick["node"].rstrip("."+storage["name"])).set(brick["memory_usage_in_bytes"])
+            storage_metrics.cpu_usage.labels(
+                brick["node"].rstrip("."+storage["name"])).set(brick["cpu_usage_in_nanoseconds"])
+
+            storage_metrics.total_number_of_containers.labels(
+                brick["node"].rstrip("."+storage["name"])).set(brick["total_number_of_containers"])
+            storage_metrics.number_of_ready_containers.labels(
+                brick["node"].rstrip("."+storage["name"])).set(brick["number_of_ready_containers"])
+
+
+@app.middleware("http")
+async def collect_metrics(request, call_next):
+    """ Collect metrics and set data to prometheus at /metrics """
+    if request.url.path == "/metrics":
+        collect_and_set_prometheus_metrics()
+
+    return await call_next(request)
+
+
+@app.get("/metrics.json")
+async def metrics_json():
+    """ Return collected metrics in JSON format at /metrics.json """
+    return collect_all_metrics()
+
+
+app.mount("/metrics", make_asgi_app())
+
+if __name__ == "__main__":
+
+    logging_setup()
+
```

Comment:
- in all `exporter` a log line stating `starting exporter` or something along those lines might be good
- yes, `uvicorn.run` might log details but explicit log line won't hurt